cl/lib/es_signal_processor.py

```diff
@@ -29,22 +41,43 @@ def updated_fields(
     :return: A list of the names of fields that have changed in the instance.
     """
     # Get the field names being tracked
+    tracked_set = None
     if es_document is AudioDocument:
         tracked_set = getattr(instance, "es_oa_field_tracker", None)
     elif es_document is ParentheticalGroupDocument:
         tracked_set = getattr(instance, "es_pa_field_tracker", None)
-    else:
+    elif es_document is PositionDocument:
         tracked_set = getattr(instance, "es_p_field_tracker", None)
 
     # Check the set before trying to get the fields
     if not tracked_set:
         return []
+
     # Check each tracked field to see if it has changed
-    changed_fields = [
-        field
-        for field in tracked_set.fields
-        if getattr(instance, field) != tracked_set.previous(field)
-    ]
+    changed_fields = []
+    for field in tracked_set.fields:
+        current_value = getattr(instance, field)
+        try:
+            # If field is a ForeignKey relation, the current value is the
+            # related object, while the previous value is the ID, get the id.
+            # See https://django-model-utils.readthedocs.io/en/latest/utilities.html#field-tracker
+            field_type = instance.__class__._meta.get_field(field)
+            if (
+                field_type.get_internal_type() == "ForeignKey"
+                and current_value
+                and "_id" not in field
```

Comment:
Would this be better: 

```suggestion
                and not field.endswith("_id")
```

Reply:
You're right. I'm changing this line! Thanks